pkg/visor/visorconfig/config.go

```diff
@@ -102,26 +78,45 @@ func defaultConfigFromCommon(cc *Common, hypervisor bool) (*V1, error) {
 		conf.Hypervisor = &config
 	}
 
+	return conf
+}
+
+// MakeDefaultConfig returns the default visor config from a given secret key (if specified).
+// The config's 'sk' field will be nil if not specified.
+// Generated config will be saved to 'confPath'.
+// This function always returns the latest config version.
+func MakeDefaultConfig(log *logging.MasterLogger, confPath string, sk *cipher.SecKey, hypervisor bool, services Services) (*V1, error) {
+	cc, err := NewCommon(log, confPath, V1Name, sk)
+	if err != nil {
+		return nil, err
+	}
+	// Enforce version and keys in 'cc'.
+	cc.Version = V1Name
+	if err := cc.ensureKeys(); err != nil {
+		return nil, err
+	}
+	// Actual config generation.
+	conf := MakeBaseConfig(cc, hypervisor, services)
+
 	return conf, nil
 }
 
 // MakeTestConfig acts like MakeDefaultConfig, however, test deployment service addresses are used instead.
-func MakeTestConfig(log *logging.MasterLogger, confPath string, sk *cipher.SecKey, hypervisor bool) (*V1, error) {
-	conf, err := MakeDefaultConfig(log, confPath, sk, hypervisor)
+func MakeTestConfig(log *logging.MasterLogger, confPath string, sk *cipher.SecKey, hypervisor bool, services Services) (*V1, error) {
+	conf, err := MakeDefaultConfig(log, confPath, sk, hypervisor, services)
 	if err != nil {
 		return nil, err
 	}
 	SetDefaultTestingValues(conf)
```

Comment:
This overwrites the Url to the `dev` ones from local no matter what is `Services` are passed.